### DISCUSSION TITLE
Applying textures

### DISCUSSION THREAD

**Discussion by josepereiroml** [pyvista/pyvista#3346] [Q&A]
Hi I am trying to apply this texture 
![patternblack](https://user-images.githubusercontent.com/65933374/191114910-ab3344ae-49e7-4326-a943-2b8992d9098c.png)
Using this object 
[object.zip](https://github.com/pyvista/pyvista/files/9602324/object.zip)
But I am getting this stretched output
![image](https://user-images.githubusercontent.com/65933374/191115126-c28e0918-3bbf-464b-a4ee-f2f54458dcf3.png)

This is the code I am using
```
import numpy as np
from scipy.interpolate import griddata
import pyvista as pv
mesh = pv.read("object.obj")
texture = pv.read_texture('patternblack.png')
mesh.subdivide(3, inplace=True)
mesh.texture_map_to_sphere(inplace=True, prevent_seam=True)
mesh.plot(texture=texture, screenshot='mapped.png')
```

Is there a way to get the texture applied and the hexagons look the same size?
![image](https://user-images.githubusercontent.com/65933374/191115419-2aa64aab-a8c3-410f-a219-7dc8145eac78.png)

Thank you!

**Comment by banesullivan** [comment]
The problem lies in using `texture_map_to_sphere` as any spherical projection is going to warp the geometry - this is a pretty classic phenomena in geospatial contexts. 

I'm not sure if I have a solution for you, but I suspect you'll have to do the texture mapping manually in order to get texture coordiantes that do not distort the image being mapped. This could prove very difficult...

Are you familiar with the concepts of texture mapping? We can help you understand the problem at least if not solve it
